test/interactions/clickInteractionTests.ts

```diff
@@ -1,160 +1,158 @@
+        assert.isTrue(callbackCalled, "callback called on clicking Component without moving mouse");
+        assert.deepEqual(lastPoint, { x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2 }, "was passed correct point");
+
+        callbackCalled = false;
+        lastPoint = null;
+        TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
+        TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH / 4, y: SVG_HEIGHT / 4}]);
+        assert.isTrue(callbackCalled, "callback called on clicking and releasing inside the Component");
+        assert.deepEqual(lastPoint, { x: SVG_WIDTH / 4, y: SVG_HEIGHT / 4 }, "was passed mouseup point");
+
+        callbackCalled = false;
+        TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
+        TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2}]);
+        assert.isFalse(callbackCalled, "callback not called if released outside Component");
+
+        callbackCalled = false;
+        TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2}]);
+        TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
+        assert.isFalse(callbackCalled, "callback not called if started outside Component");
+
+        TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
+        TestMethods.triggerFakeTouchEvent("touchmove", component.content(), [{x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2}]);
+        TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
+        assert.isTrue(callbackCalled, "callback called even if moved outside Component");
+
+        svg.remove();
+      });
+
+      it("can deregister callback using offClick()", () => {
+        let callbackWasCalled = false;
+        let callback = () => callbackWasCalled = true;
+
+        clickInteraction.onClick(callback);
+        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), 0, 0);
+        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), 0, 0);
+        assert.isTrue(callbackWasCalled, "Click interaction should trigger the callback");
+
+        callbackWasCalled = false;
+        clickInteraction.offClick(callback);
+        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), 0, 0);
+        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), 0, 0);
+        assert.isFalse(callbackWasCalled, "Callback should be disconnected from the click interaction");
+
+        svg.remove();
+      });
+
+      it("can register multiple callbacks listeners for the same Component", () => {
```

Comment:
Not sure how the `Component` is relevant -- just "can register multiple callbacks" should be sufficient.